src/db/oracledb/addresses-dao.js

```diff
@@ -1,25 +1,100 @@
+import oracledb from 'oracledb';
+
 import { parseQuery } from 'utils/parse-query';
 import { getConnection } from './connection';
 import { contrib } from './contrib/contrib';
 
+const getAddresses = async (connection, internalId, query) => {
+  const parsedQuery = parseQuery(query);
+  parsedQuery.internalId = internalId;
+
+  const { rows } = await connection.execute(contrib.getAdresses(parsedQuery), parsedQuery);
+  return rows;
+};
+
 /**
  * Queries data source for raw address data and passes it to the serializer before returning
  *
  * @param {string} internalId internal ID of person to select addresses from
  * @param {object} query Query parameters passed in with request
- * @returns {Promise<object>} Serialized address resource
+ * @returns {Promise<object>} Raw address records from data source
+ */
+const getAddressesByInternalId = async (internalId, query) => {
+  const connection = await getConnection('banner');
+  try {
+    return await getAddresses(connection, internalId, query);
+  } finally {
+    connection.close();
+  }
+};
+
+const hasSameAddressType = async (connection, internalId, addressType) => {
+  const attributes = { internalId, addressType };
+  const { rows } = await connection.execute(contrib.hasSameAddressType(), attributes);
+
+  return rows[0];
+};
+
+const phoneHasSameAddressType = async (connection, internalId, addressType) => {
+  const attributes = { internalId, addressType };
+  const { rows } = await connection.execute(contrib.phoneHasSameAddressType(), attributes);
+
+  return rows[0];
+};
+
+const updatePhoneAddrSeqno = async (connection, addrSeqno, phone) => {
+  if (phone) {
+    phone.addrSeqno = addrSeqno;
+    await connection.execute(contrib.updatePhoneAddrSeqno(), phone);
+  }
+};
+
+/**
+ * Creates address records
+ *
+ * @param {string} internalId internal ID of a person
+ * @param {object} body Request body with new address attributes
+ * @returns {Promise<object>} Raw address record from data source
  */
-const getAddressesByOsuId = async (internalId, query) => {
+const createAddress = async (internalId, body) => {
   const connection = await getConnection('banner');
   try {
-    const parsedQuery = parseQuery(query);
-    parsedQuery.internalId = internalId;
+    body.addressType = body.addressType.code;
+    body.internalId = internalId;
+    body.returnValue = { type: oracledb.DB_TYPE_VARCHAR, dir: oracledb.BIND_OUT };
+    body.seqno = { type: oracledb.DB_TYPE_VARCHAR, dir: oracledb.BIND_OUT };
+
+    // Query phone early because it is changed automatically by createAddress
+    const phone = await phoneHasSameAddressType(connection, internalId, body.addressType);
```

Comment:
We can move this to right before `updatePhoneAddrSeqno` is called. The main reason is if there's something wrong when creating an address, we can at least make sure we didn't spend time waiting for this query to be finished; it will only be called when needed.

Reply:
`createAddress` will deactivate the phone record so we must query it early.